unstructured_ingest/v2/processes/connectors/qdrant.py

```diff
@@ -0,0 +1,225 @@
+import asyncio
+import json
+import uuid
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError, WriteError
+from unstructured_ingest.utils.data_prep import batch_generator, flatten_dict
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+if TYPE_CHECKING:
+    from qdrant_client import AsyncQdrantClient
+
+
+CONNECTOR_TYPE = "qdrant"
+
+
+class QdrantAccessConfig(AccessConfig):
+    api_key: Optional[str] = Field(
+        default=None, description="API key for authentication in Qdrant Cloud."
+    )
+
+
+class QdrantConnectionConfig(ConnectionConfig):
```

Comment:
Can we actually break this up into a `qdrant-local` and `qdrant-hosted` to better isolate the auth needed for both, since I assume a lot of these parameters are only needed when connecting to the hosted version?